### DISCUSSION TITLE
NumbaWarning: The TBB threading layer requires TBB version 2019.5 or later

### DISCUSSION THREAD

**Discussion by GChims** [TDAmeritrade/stumpy#507] [Help: Coding & Implementations]
Hello all,

I get the following warning when running the Quick 1-Dimensional Exploration code on Part 10: Discovering Multidimensional Time Series Motifs (https://towardsdatascience.com/part-10-discovering-multidimensional-time-series-motifs-45da53b594bb):

```
_/usr/local/lib/python3.7/dist-packages/stumpy/stump.py:359: NumbaWarning: The TBB threading layer requires TBB version 2019.5 or later i.e., TBB_INTERFACE_VERSION >= 11005. Found TBB_INTERFACE_VERSION = 9107. The TBB threading layer is disabled.
ndist_counts = core._count_diagonal_ndist(diags, m, n_A, n_B)_
```
Although this is a warning and not en error, I wonder if this affects the output as I am not getting the motif distance output in the end. Is there something that I need to do to fix this warning?

Regards,

Gee.

**Comment by seanlaw** [comment]
@GChims Thank for your question and welcome to the STUMPY community. When you say:

>  I wonder if this affects the output as I am not getting the motif distance output in the end

In general, this warning can be ignored. It is a warning coming from `numba` and not `stumpy` per se and is just telling you that TBB is being ignored. Can you provide some more information as to the environment that you are running in and the the exact command that produces this output?

**Reply by GChims** [reply]
I get the warning when I run the below piece of code from the tutorial, in Google Colab. I am not sure if this information is sufficient
```
m = 1800
mps = {}  # Store the 1-dimensional matrix profiles
motifs_idx = {}  # Store the index locations for each pair of 1-dimensional motifs (i.e., the index location of two smallest matrix profile values within each dimension)
for dim_name in df.columns:
    mps[dim_name] = stumpy.stump(df[dim_name], m)
    motif_distance = np.round(mps[dim_name][:, 0].min(), 1)
    motifs_idx[dim_name] = np.argsort(mps[dim_name][:, 0])[:2]
    print(f"The motif pair matrix profile value in {dim_name} is {motif_distance}")
```
![Numbawarning](https://user-images.githubusercontent.com/4334133/147570552-23081eba-cf58-4d7e-8d36-8a8258dc9ccb.png)

**Reply by seanlaw** [reply]
@GChims So, what's happening is somewhat unique to Jupyter in that the printing of `numba` warning is hiding/clobbering the `print` statement within the for-loop. To overcome this and be able to see the print statements, you have three choices:

1. Add this to the top of your imports:

```
import warnings
warnings.filterwarnings('ignore')
```  

2. Since the warning disappears after the first time, simply re-run that cell in the notebook and it should clear itself out. However, you'll have to do this every time you first encounter that warning. Again, this is a `numba` warning and not a `stumpy` warning.

3. Split the commands and move the print statement to a separate cell (which circumvents the warning from clobbering the print):

```
m = 1800
mps = {}  # Store the 1-dimensional matrix profiles
motifs_idx = {}  # Store the index locations for each pair of 1-dimensional motifs (i.e., the index location of two smallest matrix profile values within each dimension)
for dim_name in df.columns:
    mps[dim_name] = stumpy.stump(df[dim_name], m)
```

then in a separate code cell

```
for dim_name in df.columns:
    motif_distance = np.round(mps[dim_name][:, 0].min(), 1)
    motifs_idx[dim_name] = np.argsort(mps[dim_name][:, 0])[:2]
    print(f"The motif pair matrix profile value in {dim_name} is {motif_distance}")
```

Please let me know if that helps!

**Reply by GChims** [reply]
Yes that helped. Thanks Sean.
